File: subprojects/frida-python/examples/open_service/dtx/sysmontap.py
import time

import frida


def on_message(message):
    print("on_message:", message)


device = frida.get_usb_device()

sysmon = device.open_service("dtx:com.apple.instruments.server.services.sysmontap")
sysmon.on("message", on_message)
sysmon.request(
    {
        "method": "setConfig:",
        "args": [
            {
                "ur": 1000,
                "cpuUsage": True,
                "sampleInterval": 1000000000,
            },
        ],
    }
)
sysmon.request({"method": "start"})
time.sleep(5)
sysmon.request({"method": "stop"})
time.sleep(1)

File: subprojects/frida-node/examples/open_service/dtx/sysmontap.ts
import frida from "frida";

let sysmon: frida.Service | null = null;

async function main() {
    const device = await frida.getUsbDevice();

    sysmon = await device.openService("dtx:com.apple.instruments.server.services.sysmontap");
    sysmon.message.connect(onMessage);
    await sysmon.request({ method: "setConfig:", args: [{ ur: 1000, cpuUsage: true, sampleInterval: 1000000000 }] });
    await sysmon.request({ method: "start" });
    await sleep(5000);
    await sysmon.request({ method: "stop" });
    await sleep(1000);
    await sysmon.cancel();
}

function onMessage(message: any) {
    console.log("onMessage:", message);
}

function sleep(duration: number) {
    return new Promise(resolve => {
        setTimeout(resolve, duration);
    });
}

main()
    .catch(e => {
        console.error(e);
    });
